app/controllers/vendors_controller.rb

```diff
@@ -23,9 +26,22 @@ def create
     end
   end
 
+  def update
+    @vendor = Vendor.find(params[:id])
+    if @vendor.approve
```

Comment:
Updated! Initially we had put in an else statement for displaying a flash notice if we did not approve something correctly. Richard pointed out that this should never actually happen, so we removed the else, but not the if. Doh!